### ISSUE TITLE
Client Discord doesn't detected via RPC

### ISSUE THREAD

**Issue by mortim59** [iCrawl/discord-vscode#77]
Hi ! I have a problem, the Rich Presence isn't detected on Visual Studio Code (I'm using Ubuntu Budgie 18.04) ,  I'm not sure that the problem is coming from Discord, because it works perfectly on Spotify

**Screen:**
https://screenshotscdn.firefoxusercontent.com/images/dc4b7aa8-baab-484a-a369-f031228324d3.png

**Comment by iCrawl** [comment]
#66

Spotify Rich Presence is an official Discord implementation so its not comparable.
Discord does a very poor job when implementing RPC for Ubuntu, nothing I can do.

**Comment by mortim59** [comment]
okey thanks

**Comment by DAKYSKYE** [comment]
> I found a solution to the extension not working on Ubuntu Budgie 18.04.
> Don't install Discord from the software centre. If you have, uninstall it.
> Go to https://discordapp.com mouse over "Download for Linux" then hit ~.deb` [[Direct Link](https://discordapp.com/api/download?platform=linux&format=deb)]
> Open your downloads folder, right click and hit "Open Tilix here" and then paste in the following
> 
> ```shell
> # This first
> dpkg -i discord-0.0.5.deb
> # If it finishes with an error basically saying it cannot finish
> # because its missing libc++1, then do the following
> sudo apt --fix-broken install
> ```
> 
> Installing the actual package and not the snap package is what made it work for me like a charm.
> Makes it seem like snap's packaging either strips features or isolates the app to the extent some stuff either doesn't work or is flaky.

Thank you @OscarXcore for solution and @iCrawl for referencing this to me <3

**Comment by OscarXcore** [comment]
No worries! I noticed that when you download the .deb off of the discord website, it opens the software centre. Anything that is installed in the software centre isn't the actual package, the .deb just links to the snap version unless you install the .deb with `dpkg`.

**Comment by thegergo02** [comment]
`sudo apt --fix-broken install` is not correct, I had to use `sudo apt install --fix-broken`

**Comment by mild-prowler** [comment]
@thegergo02 Goodness me, I'm very sorry about that. I was half-assed tired.

Fixed now.

**Comment by Stapleton** [comment]
Fixed my comment as well @thegergo02

**Comment by RyanTheDevelopan** [comment]
Um, I'm starting to experience the same problem. I'm using Windows 7 Ultimate.

**Comment by Stapleton** [comment]
Try starting discord and VSCode both as Administrator

**Comment by Blaze135** [comment]
I have problem with connect to discord
https://gyazo.com/8a41e2bf56965135aaf5170d9c1db808

**Comment by Pverte** [comment]
Hello, I'm sorry but I have the same problem with the extension : I said me that Couldn't connect to Discord via RPC: Error: Could not connect.
I'm on Fedora 31, Is that anyone have an idea ?

**Comment by C0braD3v** [comment]
I'd like to stress the comment made [here](https://github.com/iCrawl/discord-vscode/issues/77#issuecomment-443592725). Basic fix: Don't use Software Installer. For Discord **OR** vscode.

**Comment by Pverte** [comment]
It’s ok for me, I downloaded manually and now it’s work

**Comment by danbulant** [comment]
> Don't use Software Installer

Don't forget that this is an issue with snap based installers (like in Ubuntu). Other software centers which don't use snaps usually uses the `deb` package, meaning the installation will work normally.

Snaps are pretty isolated from system, nearly like running a VM. They're more secure, but also more limited and some features like rich presence might not work.

**Comment by iDevoid** [comment]
has anyone faced the same problem on Fedora?

**Comment by kavin25** [comment]
I am on a Mac. How do I solve the issue?

**Comment by kavin25** [comment]
> I am on a Mac. How do I solve the issue? I downloaded the app from their website

**It ended up connecting after I reinstalled the extension**

**Comment by RCout1nho** [comment]
i'm on ubuntu 20.04 and i have the rpc connection problem, anyone can help me?

**Comment by AliRedaHajj** [comment]
I have the same error but I am on windows 10 home

**Comment by RCout1nho** [comment]
> i'm on ubuntu 20.04 and i have the rpc connection problem, anyone can help me?

I resolved it, i just installed vscode and discord from oficial sites and install with dpkg

**Comment by whatland** [comment]
Just restarting my pc worked for me.

**Comment by Russhvith** [comment]
When i install the discord presence and start editing it does not come in Discord Status

**Comment by dinogomez** [comment]
> i'm on ubuntu 20.04 and i have the rpc connection problem, anyone can help me?

Try following @Stapleton 's Fix
```# This first
dpkg -i discord-0.0.5.deb
# If it finishes with an error basically saying it cannot finish
# because its missing libc++1, then do the following
sudo apt install --fix-broken```

**Comment by dinogomez** [comment]
@RCout1nho  you can uninstall pre-installed discord by using this command. 
```sudo snap remove discord```

**Comment by ElnuDev** [comment]
TL;DR: If this isn't working for you:

1. If you installed Discord through Ubuntu Software, uninstall it.
2. Reinstall Discord with the download directly from Discord's website.
3. If you installed Visual Studio Code through Ubuntu Software, uninstall it.
4. Reinstall Visual Studio Code directly from Visual Studio's website.
5. If you get the error "Failed to install file: not supported" from Ubuntu Software when installing Discord or Visual Studio Code directly from their websites, use the GDebi Package Installer instead. It usually installs properly when Ubuntu Software fails, and when it fails it gives a detailed error.

**Comment by developersd1382** [comment]
it was helpful tahnks

**Comment by HarshitKumarOjha** [comment]
I uninstalled both discord and vs code from ubuntu software 
and then installed them from the official websites
It worked for me..Thanks :)

**Comment by Avnoor-Gamerz** [comment]
> I found a solution to the extension not working on Ubuntu Budgie 18.04.
> Don't install Discord from the software centre. If you have, uninstall it.
> Go to https://discordapp.com mouse over "Download for Linux" then hit ~.deb` [[Direct Link](https://discordapp.com/api/download?platform=linux&format=deb)]
> Open your downloads folder, right click and hit "Open Tilix/(some terminal emulator) here" and then paste in the following
> 
> ```shell
> # This first
> dpkg -i discord-0.0.5.deb
> # If it finishes with an error basically saying it cannot finish
> # because its missing libc++1, then do the following
> sudo apt install --fix-broken
> ```
> 
> Installing the actual package and not the snap package is what made it work for me like a charm.
> Makes it seem like snap's packaging either strips features or isolates the app to the extent some stuff either doesn't work or is flaky.

what will for windows

**Comment by Stapleton** [comment]
> what will for windows

Run them as admin. A proper solution for windows should be made into a new issue.

**Comment by Th3Ph4nt0m** [comment]
> I found a solution to the extension not working on Ubuntu Budgie 18.04.
> Don't install Discord from the software centre. If you have, uninstall it.
> Go to https://discordapp.com mouse over "Download for Linux" then hit ~.deb` [[Direct Link](https://discordapp.com/api/download?platform=linux&format=deb)]
> Open your downloads folder, right click and hit "Open Tilix/(some terminal emulator) here" and then paste in the following
> 
> ```shell
> # This first
> dpkg -i discord-0.0.5.deb
> # If it finishes with an error basically saying it cannot finish
> # because its missing libc++1, then do the following
> sudo apt install --fix-broken
> ```
> 
> Installing the actual package and not the snap package is what made it work for me like a charm.
> Makes it seem like snap's packaging either strips features or isolates the app to the extent some stuff either doesn't work or is flaky.

Thank you soooo much!

I have a similar solution:

1. Open Discord
2. Remove Discord via "Ubuntu Software" (While Discord is still opened!)
3. Wait until the store is finished
4. Close Discord
5. Open Discord again

This worked for me!

**Comment by Peresh** [comment]
For OS X Big Sur, grant the vscode "Full Disk Access" may help.

Just go Sustem Preferences -> Security & Privacy -> Full Disk Access -> Add the vscode

**Comment by Avnoor-Gamerz** [comment]
Ok

**Comment by iVictory2004** [comment]
For anybody that it doesn't work for: I got it working, by just uninstalling vscode and installing it back,

**BEWARE THOUGH, AS IT WILL DELETE ALL YOUR SETTINGS AND EXTENSIONS, IT'S GONNA BE LIKE BEFORE YOU INSTALLED IT AND WENT THROUGH THE COSTUMIZATION PROCESS, VERY ANNOYING.**



1. Uninstall VsCode
2. Install It back Up
3. Install the [rich presence plugin](https://marketplace.visualstudio.com/items?itemName=icrawl.discord-vscode) again.
4. You should be done I hope

 _NOT DISCORD_, but Vscode,

**Comment by DiegoNG90** [comment]
> I found a solution to the extension not working on Ubuntu Budgie 18.04. Don't install Discord from the software centre. If you have, uninstall it. Go to https://discordapp.com mouse over "Download for Linux" then hit ~.deb` [[Direct Link](https://discordapp.com/api/download?platform=linux&format=deb)] Open your downloads folder, right click and hit "Open Tilix/(some terminal emulator) here" and then paste in the following
> 
> ```shell
> # This first
> dpkg -i discord*.deb
> # If it finishes with an error basically saying it cannot finish
> # because its missing libc++1, then do the following
> sudo apt install --fix-broken
> ```
> 
> Installing the actual package and not the snap package is what made it work for me like a charm. Makes it seem like snap's packaging either strips features or isolates the app to the extent some stuff either doesn't work or is flaky.

It worked perfectly for me.
Danke schön @kalkafox

**Comment by DexrnZacAttack** [comment]
What about people that use Fedora, as there seems to be no RPM package and updating Discord is hella annoying, so Flatpak is the easiest way.
